packages/suite/src/components/wallet/TokenIconSetWrapper.tsx

```diff
@@ -0,0 +1,44 @@
+import {
+    selectCoinDefinitions,
+    getContractAddressForNetwork,
+} from '@suite-common/token-definitions';
+import {
+    enhanceTokensWithRates,
+    getTokens,
+    sortTokensWithRates,
+} from 'src/utils/wallet/tokenUtils';
+import { TokenIconSet } from '@trezor/product-components';
+import { useSelector } from 'src/hooks/suite';
+import { selectLocalCurrency } from 'src/reducers/wallet/settingsReducer';
+import { selectCurrentFiatRates } from '@suite-common/wallet-core';
+import { Account } from '@suite-common/wallet-types';
+
+type TokenIconSetWrapperProps = {
+    account: Account;
+};
+
+export const TokenIconSetWrapper = ({ account }: TokenIconSetWrapperProps) => {
+    const { tokens: accountTokens, symbol: network } = account;
+    const fiatRates = useSelector(selectCurrentFiatRates);
+    const coinDefinitions = useSelector(state => selectCoinDefinitions(state, network));
+    const localCurrency = useSelector(selectLocalCurrency);
+    const tokensWithRates = enhanceTokensWithRates(
+        accountTokens,
+        localCurrency,
+        network,
+        fiatRates,
+    );
+    const sortedTokens = tokensWithRates.sort(sortTokensWithRates);
+    const tokens = getTokens(sortedTokens, network, coinDefinitions);
+    const contractAddresses = tokens.shownWithBalance
```

Comment:
instead of passing both 
```
            tokens={tokens.shownWithBalance}
            contractAddresses={contractAddresses}
```

I would rather see something like
```
    const tokens = getTokens(sortedTokens, network, coinDefinitions);

    // Enhance each token with the contract address
    const tokensWithEnhancedAddress = tokens.shownWithBalance.map(token => ({
        ...token,
        enhancedContractAddress: getContractAddressForNetwork(network, token.contract),
    }));
```

BUT another solution, and probably better, can be to move `getContractAddressForNetwork` to `wallet-utils` package so that you can import it directly in your component

Reply:
I prefer to move it to another place to avoid the build up of hacks, moved it as well as the tests, let's see how the pipeline reacts to it